trajectory_tracker/include/trajectory_tracker/path2d.h

```diff
@@ -93,7 +93,7 @@ class Path2D : public std::vector<Pose2D>
       const float angle_pose = allow_backward_motion ? it->yaw_ : angle;
       const float sign_vel_req = std::cos(angle) * std::cos(angle_pose) + std::sin(angle) * std::sin(angle_pose);
       if (sign_vel_prev * sign_vel_req < 0)
-        return it + 1;
```

Comment:
`it` is already next to the switching back point.

Reply:
It's also covered by TrajectoryTrackerTest/SwitchBack.
Adding separate unit test for it is good.